os-server/src/main/java/app/openschool/coursemanagement/api/mapper/CourseOfUserMapper.java

```diff
@@ -0,0 +1,64 @@
+package app.openschool.coursemanagement.api.mapper;
+
+import app.openschool.coursemanagement.api.dto.CourseOfUserDto;
+import app.openschool.coursemanagement.entity.Course;
+import app.openschool.coursemanagement.entity.Module;
+import app.openschool.coursemanagement.entity.ModuleItem;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+public class CourseOfUserMapper {
+
+  public static List<CourseOfUserDto> toCourseOfUserDtoList(List<Course> courseList) {
+    List<CourseOfUserDto> courseOfUserDtoList = new ArrayList<>();
+    for (Course course : courseList) {
+      courseOfUserDtoList.add(toCourseOfUserDto(course));
+    }
+    return courseOfUserDtoList;
+  }
+
+  public static CourseOfUserDto toCourseOfUserDto(Course course) {
+
+    String courseStatus = course.getStatus().getType();
+    String buttonName = courseStatus.equals("COMPLETED") ? "RATE COURSE" : "RESUME COURSE";
```

Comment:
You should not think about the rendering in backend code, just return the status, FE will decide what to render.